src/ability/msg.rs

```diff
@@ -0,0 +1,183 @@
+use crate::{ability::traits::Command, prove::TryProve};
+use libipld_core::ipld::Ipld;
+use std::{collections::BTreeMap, str::FromStr};
+use url::Url;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Msg {
+    to: Url,
+    from: Url,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct MsgSend {
+    to: Url,
+    from: Url,
+    message: String,
+}
+
+// TODO is the to or from often also the subject? Shoudl that be accounted for?
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct MsgReceive {
+    to: Url,
+    from: Url,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
```

Comment:
as per your video, we can remove these via https://crates.io/crates/optional_struct, so you can generate optional versions.

Reply:
You mean https://docs.rs/derive_builder/latest/derive_builder/? Yeah, I'd say the latter is more explicit about the Option version. `derive_builder` actually has different use cases. So, I'd say it's about intention.